### ISSUE TITLE
Connect command doesn't work (possible Python error)

### ISSUE THREAD

**Issue by KamicDemon** [seemoo-lab/internalblue#21]
Hello,
First I want to express my thanks to everyone that has helped support this awesome project. I'm so excited to try this, but I'm not very experienced so it's possible I'm making a lot of rookie mistakes.

I'm using a rooted Nexus 5 running Android 6.0.1. Build number is M4B30Z

I followed the instructions detailed in the android_bluetooth_stack README, using the pre-compiled android6_0_1. I did no building of my own. I ran the "git apply" command before pushing the .so file and running the other adb commands.

I installed the h4bcm_wireshark_dissector following the instructions on that git page.

I didn't encounter any errors while following those steps so I assume I did them right.

USB debugging and HCI snoop logs are both enabled.

When I run internalblue, I can open wireshark through "monitor start" after selecting my adb interface. I don't see any LMP traffic when initiating a pair request or scan from the Nexus5.

When I try to connect using the internalblue "connect" command I get errors and the program exits. Screenshots provided below.

Once again, I'm sorry if this is a total newbie mistake, I'm just not sure what's going on.

![Screenshot from 2020-04-06 04-57-52](https://user-images.githubusercontent.com/37749338/78524482-0f9cd380-780f-11ea-8de2-0b7018935638.png)
![Screenshot from 2020-04-06 04-06-46](https://user-images.githubusercontent.com/37749338/78524493-17f50e80-780f-11ea-95a8-a770d5af3d8b.png)

**Comment by fmagin** [comment]
That is definitely some type error on our side, looking into it.

**Comment by fmagin** [comment]
At least not an issue in current `master`.
On current `master` the address is converted with `addr = bytearray.fromhex(addr)` in `parse_bt_addr` at line 122.

That code hasn't changed since January 2019 either.
How did you install internalblue?

**Comment by KamicDemon** [comment]
I cloned the repo, changed into the directory, and did
"sudo python3 setup.py install"
I think it was an issue with the machine I was using earlier... I made a fresh linux VM on another machine and gave it a try and it seems to be working a bit better. I can use the "connect" command without any issues, but I'm still not seeing LMP traffic.

I see lots of "Rcvd Unknown HCI Packet Type 0x07" messages, as you can see in the screenshot. I'm reading other issues to try and find a fix.

![Screenshot_2020-04-06_03-45-08](https://user-images.githubusercontent.com/37749338/78615252-fd3a8e00-783e-11ea-9330-8d595b00e8e6.png)

**Comment by jiska2342** [comment]
Hi,

`Rcvd Unknown HCI Packet Type 0x07` is when your Wireshark dissector plugin doesn't work, that's the LMP packets. The dissector works in all versions of Wireshark 3.* I tried so far, but I encountered that the home directory changed in the most recent version of my machine and I needed to copy it to a different plugin subfolder to get it working.

I don't have any issues with the `connect` command, though, and it also works well in your screenshot.

**Comment by fmagin** [comment]
My guess would be that `python3 setup.py install` doesn't upgrade packages, at least `pip install ./` doesn't unless you add `--upgrade`

**Comment by KamicDemon** [comment]
I figured out the problem!
I double checked a few things in wireshark:
Help -> About Wireshark: Told me the exact version I was using
In that same "About Wireshark" Page there's a tab named "Folders" that lists both the global and personal plugin files Wireshark is sourcing its plugins from. I noticed that they were different than the one the h4bcm.so file was installed to.

Apparently, when running make install, a "plugins/3.0" folder was created to host the h4bcm.so file, but my Wireshark (Version 3.2.2) was sourcing from a "plugins/3.2" folder. The personal version of this directory didn't even exist, as i show in the attached screenshot.

Clicking on the link in the "About Wireshark" window prompted me to make the folder, which I did, and then copied and pasted the h4bcm.so file into **both** the global and personal plugin folders just to be safe - and it all works! Screenshots attached.

I hope this can help someone else if they encounter the same problem (although we've sort of drifted from the topic of in the issue title)

Closing the issue, thanks again for all the help :)
![Screenshot_2020-04-07_06-55-33](https://user-images.githubusercontent.com/37749338/78732101-a76c0000-797c-11ea-8b95-f0e8b8e8ba74.png)
![creating_v3 2_plugin_directory](https://user-images.githubusercontent.com/37749338/78732110-acc94a80-797c-11ea-9a6e-cd0924374b8e.png)
![moving_h4bcm_to_personal_and_global_plugin_folders](https://user-images.githubusercontent.com/37749338/78732113-af2ba480-797c-11ea-86c7-1ba28784832d.png)
![LMP_works](https://user-images.githubusercontent.com/37749338/78732116-b18dfe80-797c-11ea-9d15-6fcbe5526a7a.png)
